### PR TITLE
Closes #262: Allow for package-wide config defaults

### LINKED ISSUE #262: Package-wide config
Currently the config options for the evaluations are only pulled in on running the report. It would be good to have package-wide access to the config options, so that individual computations have access to the default options if they are imported.

### PR DESCRIPTION
**Pull request by dzeber**
- There is now a package-level config object `global_config`. It is initialized to the package default options but can be updated by the user.
- Config can be overridden at the report and evaluation level (ie. within those class instances). These overrides do not modify the global config, but changes to the global config will get pulled in. Eg. if a global config option changes that is not set in the report config, the report will pull in the new value.
- Config management is outsourced to [`confuse`](https://github.com/beetbox/confuse), which handles merging layers and type-checking.

Hope this doesn't seem overkill :) I'm not sure yet what all of the use cases will be, and I want to keep it flexible for now. Two main use cases I do want to support are running a static report (could be as a scheduled job), and interactive use of the evaluations, eg. in a notebook as exploratory data analysis.

This will all be documented in soon-to-be-added user-guide documentation.

### CODE REVIEW

File: presc/report/runner.py [hunk 1]
```diff
@@ -75,30 +54,40 @@ def _updated_jb_toc(report_config):
...
+                return {k: v for k, v in filtered.items() if v}
+        else:
+            return toc_part
+
+    toc = yaml.load(toc_str, Loader=yaml.FullLoader)
+    # Drop excluded file entries from the TOC and any subsections that are left empty.
+    filtered_toc = drop_file_entries(toc, incl_pages)
+    # Drop any top-level entries that don't have files in their hierarchy
+    filtered_toc = [x for x in filtered_toc if "file" in str(x)]
+    return yaml.dump(filtered_toc)
```
Comment (gleonard-m) [inline presc/report/runner.py:90]: This seems overly complicated.  There is not way to to apply the include and exclude lists directly to the toc?

[the remark this answers is not in the document]
Comment (dzeber) [reply presc/report/runner.py:90]: You're right, there is in fact a way to do it directly that I totally overlooked in the docs. I will update this.

File: tests/test_runner.py [hunk 2]
```diff
@@ -12,43 +11,49 @@
...
+  evaluations_exclude: "*"
+"""
 
 
 @pytest.fixture
-def config_override_report(config_default, config_report):
-    conf = deepcopy(config_default)
-    conf["report"] = config_report
+def config_report():
+    conf = PrescConfig(global_config)
```
[the remark this answers is not in the document]
Comment (dzeber) [reply tests/test_runner.py:46]: If no parameter is passed, it creates a new config object that is unconnected to the global_config. Passing it in means later changes in the global_config are picked up.

File: presc/evaluations/conditional_metric.py [hunk 3]
```diff
@@ -134,24 +145,30 @@ def compute_for_column(self, colname, metric, **kwargs):
...
+        try:
+            col_overrides = col_overrides.get()
+        except ConfigError:
+            col_overrides = None
         if col_overrides:
-            comp_config.update(col_overrides.get(colname, {}))
-        comp_config.update(kwargs)
+            comp_config.set(col_overrides)
+        if kwargs:
+            comp_config.set(kwargs)
```
[the remark this answers is not in the document]
Comment (dzeber) [reply presc/evaluations/conditional_metric.py:162]: Good point. I will do this in the next refactoring PR.

File: presc/report/runner.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gleonard-m) [reply presc/report/runner.py]: Much better!

**Pull request merged**